src/daemons/GraphDaemon.cpp

```diff
@@ -43,6 +46,11 @@ extern Status setupBreakpad();
 DECLARE_string(flagfile);
 DECLARE_bool(containerized);
 
+enum ServiceStatus { STATUS_UNINITIALIZED = 0, STATUS_RUNNING = 1, STATUS_STOPPED = 2 };
+std::atomic<ServiceStatus> serverStatus_{STATUS_UNINITIALIZED};
+std::mutex muStop_;
+std::condition_variable cvStop_;
```

Comment:
Prefer to provide a class to handle this signal process

Reply:
done